File: src/main/java/edu/kit/iti/formal/stvs/view/STVSRootView.java
package edu.kit.iti.formal.stvs.view;

import edu.kit.iti.formal.stvs.view.editor.EditorPane;
import edu.kit.iti.formal.stvs.view.menu.STVSMenuBar;
import javafx.scene.control.TabPane;
import javafx.scene.layout.Pane;

/**
 * Created by csicar on 09.01.17.
 */
public class STVSRootView extends Pane {
  private STVSMenuBar menuBar;
  private EditorPane editor;
  private TabPane specifications;

  public STVSMenuBar getMenuBar() {
    return menuBar;
  }

  public EditorPane getEditor() {
    return editor;
  }

  public TabPane getSpecifications() {
    return specifications;
  }

  public void setMenuBar(STVSMenuBar menuBar) {
    this.menuBar = menuBar;
  }

  public void setEditor(EditorPane editor) {
    this.editor = editor;
  }

  public void setSpecifications(TabPane specifications) {
    this.specifications = specifications;
  }
}

File: stvs/src/main/kotlin/edu/kit/iti/formal/stvs/view/StvsRootView.kt
package edu.kit.iti.formal.stvs.view

import edu.kit.iti.formal.stvs.view.editor.EditorPane
import edu.kit.iti.formal.stvs.view.spec.SpecificationsPane
import javafx.scene.control.SplitPane

/**
 * This view holds the editor and the specifications pane.
 *
 * @author Carsten Csiky
 */
class StvsRootView(ed: EditorPane, var specifications: SpecificationsPane) : SplitPane() {
    var editor: EditorPane = ed
        set(value) {
            field = editor; items[0] = value
        }


    /**
     * This creates an instance that holds an editor and the specifications pane.
     *
     * @param editor Editor to display
     * @param specifications Pane to display
     */
    init {
        ViewUtils.setupClass(this)
        // for presentations
        //this.setStyle("-fx-font-size: 16pt");
        items.addAll(editor, specifications)
    }
}
